script.js

```diff
@@ -1,4 +1,214 @@
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length); // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex]; // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex]; // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard; // Increment currentIndex to shuffle the next pair of cards
+    currentIndex += 1;
+  } // Return the shuffled deck
+  return cardDeck;
+};
+var deck = shuffleCards(makeDeck());
+
+//Global Variables
+//Default Mode is player turn
+var currentPlayer = "playerTurn";
+//Player & Computer cards in an array
+playerCards = [];
+computerCards = [];
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  currentPlayer = "playerTurn";
+  var deck = shuffleCards(makeDeck());
+  var playerCounter = 0;
+  var computerCounter = 0;
+
+  //Deal 2 Cards each with deck.pop
+  while (playerCounter < 2) {
+    playerCard = deck.pop();
+    playerCards.push(playerCard);
+    playerCounter = playerCounter + 1;
+  }
+  //If player has drawn two cards, it is computer's turn to draw two cards
+  if (playerCards.length == 2) {
+    currentPlayer = "computerTurn";
+    while (computerCounter < 2) {
+      computerCard = deck.pop();
+      computerCards.push(computerCard);
+      computerCounter = computerCounter + 1;
+    }
+    //Change the rank of J,Q,K to be 10
+    if (
+      playerCards[0].name == "King" ||
+      playerCards[0].name == "Queen" ||
+      playerCards[0].name == "Jack"
+    ) {
+      playerCards[0].rank = 10;
+    }
+    if (
+      playerCards[1].name == "King" ||
+      playerCards[1].name == "Queen" ||
+      playerCards[1].name == "Jack"
+    ) {
+      playerCards[1].rank = 10;
+    }
+    if (
+      computerCards[0].name == "King" ||
+      computerCards[0].name == "Queen" ||
+      computerCards[0].name == "Jack"
+    ) {
+      computerCards[0].rank = 10;
+    }
+    if (
+      computerCards[1].name == "King" ||
+      computerCards[1].name == "Queen" ||
+      computerCards[1].name == "Jack"
+    ) {
+      computerCards[1].rank = 10;
+    }
+    //Ace is 11 when Only 2 cards are drawn
+    if (playerCards.length == 2) {
+      if (playerCards[0].name == "Ace") {
+        playerCards[0].rank = 11;
+      }
+      if (playerCards[1].name == "Ace") {
+        playerCards[1].rank = 11;
+      }
+      if (computerCards.length == 2) {
+        if (computerCards[0].name == "Ace") {
+          computerCards[0].rank = 11;
+        }
+        if (computerCards[1].name == "Ace") {
+          computerCards[1].rank = 11;
+        }
+      }
```

Comment:
all this could have been modifies in the makeDeck function

Reply:
^^